src/constructSignatureString.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const calculateCrypto = require('./calculateCrypto')
+
+const constructSignatureString = (req, options) => {
+  const {
+    getSignatureEncoding = () => 'base64',
+    getAlgorithm = require('./getAlgorithm'),
+    sharedSecret
+  } = options
+
+  if (!sharedSecret) {
+    throw new Error('Missing shared secret in options')
+  }
+
+  return calculateCrypto({
+    method: 'hmac',
+    algorithm: getAlgorithm(req, options),
+    sharedSecret: sharedSecret,
+    message: getMessage(req, options),
+    encoding: getSignatureEncoding(req, options)
+  })
+}
+
+const getMessage = (req, options) => {
+  const { signedHeaders, created, expires } = options
+  const { headers, method, url } = req
+  const parsedUrl = new URL(url)
+
+  return signedHeaders
+    .map((signatureHeader) => {
+      switch (signatureHeader) {
```

Comment:
Could we provide compatibility with pascal case headers (ie. `Content-Type`)?
```js
 switch (signatureHeader.toLowerCase()) {
```

Reply:
Good idea. I added this suggestion and also formatted the header keys input by the request to all be lower case. This should cover all combinations of case mismatch.